client/softlayer_client_test.go

```diff
@@ -35,6 +35,16 @@ var _ = Describe("SoftLayerClient", func() {
 		})
 	})
 
+	Context("#NewSoftLayerClient_HTTPClient", func() {
```

Comment:
Yes. This is good. However, I'd expect another `It()` that passes your own `HTTPClient` and verifies that it is indeed used. If you add that should be good enough coverage. Thanks.